addons/medical/functions/fnc_treatmentAdvanced_bandageLocal.sqf

```diff
@@ -94,12 +94,73 @@ if (_impact > 0 && {GVAR(enableAdvancedWounds)}) then {
     [_target, _impact, _part, _mostEffectiveSpot, _mostEffectiveInjury, _bandage] call FUNC(handleBandageOpening);
 };
 
-// If all wounds have been bandaged, we will reset all damage to 0, so the unit is not showing any blood on the model anymore.
-if (GVAR(healHitPointAfterAdvBandage) && {{(_x select 2) == _part && {((_x select 4) * (_x select 3)) > 0}}count _openWounds == 0}) then {
-    _hitSelections = ["head", "body", "hand_l", "hand_r", "leg_l", "leg_r"];
-    _hitPoints = ["HitHead", "HitBody", "HitLeftArm", "HitRightArm", "HitLeftLeg", "HitRightLeg"];
-    _point = _hitPoints select (_hitSelections find _selectionName);
-    _target setHitPointDamage [_point, 0];
+// If all wounds to a body part have been bandaged, reset damage to that body part to zero
+// so that the body part functions normally and blood is removed from the uniform.
+// Arma combines left and right arms into a single body part (HitHands), same with left and right legs (HitLegs).
+// Arms are actually hands.
+if (GVAR(healHitPointAfterAdvBandage)) then {
+    private["_currentWounds", "_headWounds", "_bodyWounds", "_legsWounds", "_armWounds"];
+
+    // Get the list of the wounds the target is currently suffering from.
+    _currentWounds = _target getVariable [QGVAR(openWounds), []];
+
+    // Tally of unbandaged wounds to each body part.
+    _headWounds = 0;
+    _bodyWounds = 0;
+    _legsWounds = 0;
+    _armWounds  = 0;
+
+    // Loop through all current wounds and add up the number of unbandaged wounds on each body part.
+    {
+        _x params ["", "", "_bodyPart", "_numOpenWounds", "_bloodLoss"];
+
+        // Head
+        if (_bodyPart == 0 && {(_numOpenWounds * _bloodLoss) > 0}) then {
```

Comment:
Create an early exit condition for `(_numOpenWounds * _bloodLoss) > 0` that way it only has to be evaluated 1 time instead of 6 times in the worst case, as well as skipping 6 value checks and making the code look cleaner.